samples/client/petstore/swift5/vaporLibrary/Sources/PetstoreClient/APIs/StoreAPI.swift

```diff
@@ -0,0 +1,228 @@
+
+        guard let apiClient = Configuration.apiClient else {
+            fatalError("Configuration.apiClient is not set.")
+        }
+
+        return apiClient.send(.GET, headers: headers, to: URI(string: URLString)) { request in
+            try Configuration.apiWrapper(&request)
+            
+            
+            
+            try beforeSend(&request)
+        }
+    }
+
+    public enum GetOrderById {
+        case http200(value: Order, raw: ClientResponse)
+        case http400(value: Void, raw: ClientResponse)
+        case http404(value: Void, raw: ClientResponse)
+        case http0(value: Order, raw: ClientResponse)
+    }
+
+    /**
+     Find purchase order by ID
+     GET /store/order/{order_id}
+     For valid response try integer IDs with value <= 5 or > 10. Other values will generated exceptions
+     - parameter orderId: (path) ID of pet that needs to be fetched 
+     - returns: `EventLoopFuture` of `GetOrderById` 
+     */
+    open class func getOrderById(orderId: Int64, headers: HTTPHeaders = PetstoreClient.customHeaders, beforeSend: (inout ClientRequest) throws -> () = { _ in }) -> EventLoopFuture<GetOrderById> {
+        return getOrderByIdRaw(orderId: orderId, headers: headers, beforeSend: beforeSend).flatMapThrowing { response -> GetOrderById in
+            switch response.status.code {
+            case 200:
+                return .http200(value: try response.content.decode(Order.self, using: Configuration.contentConfiguration.requireDecoder(for: Order.defaultContentType)), raw: response)
+            case 400:
+                return .http400(value: (), raw: response)
+            case 404:
+                return .http404(value: (), raw: response)
+            default:
+                return .http0(value: try response.content.decode(Order.self, using: Configuration.contentConfiguration.requireDecoder(for: Order.defaultContentType)), raw: response)
+            }
+        }
+    }
+
+
+    /**
+     Place an order for a pet
+     POST /store/order
+     - parameter body: (body) order placed for purchasing the pet 
+     - returns: `EventLoopFuture` of `ClientResponse` 
+     */
+    open class func placeOrderRaw(body: Order, headers: HTTPHeaders = PetstoreClient.customHeaders, beforeSend: (inout ClientRequest) throws -> () = { _ in }) -> EventLoopFuture<ClientResponse> {
+        let path = "/store/order"
+        let URLString = PetstoreClient.basePath + path
+
+        guard let apiClient = Configuration.apiClient else {
+            fatalError("Configuration.apiClient is not set.")
+        }
+
+        return apiClient.send(.POST, headers: headers, to: URI(string: URLString)) { request in
+            try Configuration.apiWrapper(&request)
+            
+            
+            try request.content.encode(body, using: Configuration.contentConfiguration.requireEncoder(for: Order.defaultContentType))
+            
+            try beforeSend(&request)
+        }
+    }
+
+    public enum PlaceOrder {
```

Comment:
@aymanbagabas looking at this enum I'm not sure if I feel confident about it.
I see the reason on why you did it this way, but I'm afraid if this will be a restrictive API to use in the future, and maybe a class response would be better.
Looking at this it is not obvious to someone that doesn't know the http status code if it succeeded or not.
With the other libraries, it is obvious if the request was successful or not.